### ISSUE TITLE
Add OIDC support

### ISSUE THREAD

**Issue by SuperSandro2000** [majewsky/portunus#24]
Right now we are using dex-idp to support OIDC/OAuth2 but that has no function to properly remember sessions with support for logging our or select scopes to grant.
It would probably be a lot easier if portunus would just add an OAuth2 endpoint and clients would be statically configured.

**Comment by majewsky** [comment]
I agree. I will probably not have the time to work on this in the short or midterm, but if anyone wants to work on it, you have my support. Please keep in touch and prefer to submit PRs early so we can align on the design and structure.

**Comment by SuperSandro2000** [comment]
I fixed the remember session patch for dex idp, so at least we have some half backed solution for this https://github.com/SuperSandro2000/dex/commit/cfcd6ac9f64dea0b068314048d2c534149f6c925

Still you can't select scopes or log out without clearing cookies.

**Comment by majewsky** [comment]
Most likely, we will use something like https://github.com/ory/hydra and only implement the UI part (what Hydra calls "login and consent app").
